src/net/src/tls_tcp/mod.rs

```diff
@@ -0,0 +1,403 @@
+            #[cfg(feature = "boringssl")]
+            TlsTcpStreamImpl::Boringssl(s) => s.do_handshake(),
+            #[cfg(feature = "openssl")]
+            TlsTcpStreamImpl::Openssl(s) => s.do_handshake(),
+        }
+    }
+
+    pub fn shutdown(&mut self) -> Result<ShutdownResult> {
+        match &mut self.inner {
+            #[cfg(feature = "boringssl")]
+            TlsTcpStreamImpl::Boringssl(s) => s.shutdown().map(|v| v.into()),
+            #[cfg(feature = "openssl")]
+            TlsTcpStreamImpl::Openssl(s) => s.shutdown().map(|v| v.into()),
+        }
+    }
+}
+
+impl Read for TlsTcpStream {
+    fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
+        match &mut self.inner {
+            #[cfg(feature = "boringssl")]
+            TlsTcpStreamImpl::Boringssl(s) => s.read(buf),
+            #[cfg(feature = "openssl")]
+            TlsTcpStreamImpl::Openssl(s) => s.read(buf),
+        }
+    }
+}
+
+impl Write for TlsTcpStream {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        match &mut self.inner {
+            #[cfg(feature = "boringssl")]
+            TlsTcpStreamImpl::Boringssl(s) => s.write(buf),
+            #[cfg(feature = "openssl")]
+            TlsTcpStreamImpl::Openssl(s) => s.write(buf),
+        }
+    }
+
+    fn flush(&mut self) -> Result<()> {
+        match &mut self.inner {
+            #[cfg(feature = "boringssl")]
+            TlsTcpStreamImpl::Boringssl(s) => s.flush(),
+            #[cfg(feature = "openssl")]
+            TlsTcpStreamImpl::Openssl(s) => s.flush(),
+        }
+    }
+}
+
+impl event::Source for TlsTcpStream {
+    fn register(&mut self, registry: &Registry, token: Token, interest: Interest) -> Result<()> {
+        match &mut self.inner {
+            #[cfg(feature = "boringssl")]
+            TlsTcpStreamImpl::Boringssl(s) => s.register(registry, token, interest),
+            #[cfg(feature = "openssl")]
+            TlsTcpStreamImpl::Openssl(s) => s.register(registry, token, interest),
+        }
+    }
+
+    fn reregister(
+        &mut self,
+        registry: &mio::Registry,
+        token: mio::Token,
+        interest: mio::Interest,
+    ) -> Result<()> {
+        match &mut self.inner {
+            #[cfg(feature = "boringssl")]
+            TlsTcpStreamImpl::Boringssl(s) => s.reregister(registry, token, interest),
+            #[cfg(feature = "openssl")]
+            TlsTcpStreamImpl::Openssl(s) => s.reregister(registry, token, interest),
+        }
+    }
+
+    fn deregister(&mut self, registry: &mio::Registry) -> Result<()> {
+        match &mut self.inner {
+            #[cfg(feature = "boringssl")]
+            TlsTcpStreamImpl::Boringssl(s) => s.deregister(registry),
+            #[cfg(feature = "openssl")]
+            TlsTcpStreamImpl::Openssl(s) => s.deregister(registry),
+        }
+    }
+}
+
+pub struct TlsTcpAcceptor {
+    inner: TlsTcpAcceptorImpl,
+}
+
+enum TlsTcpAcceptorImpl {
+    Boringssl(boringssl::TlsTcpAcceptor),
```

Comment:
Shouldn't this be wrapped in a `#[cfg(feature = "boringssl")]`?

Reply:
Yes. Fixed and added powerset check to CI.